library/Vanilla/VisibleTextLengthValidator.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @author Patrick Kelly <patrick.k@vanillaforums.com>
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla;
+
+use Vanilla\Invalid;
+use Vanilla\Formatting\FormatService;
+use Vanilla\Contracts\LocaleInterface;
+
+/*
+ * Validates the Body field length by stripping any formatting code.
+ */
+class VisibleTextLengthValidator {
+
+    /** @var \Vanilla\Contracts\LocaleInterface */
+    private $locale;
+
+    /** @var \Vanilla\Formatting\FormatService */
+    private $formatService;
+
+    /** @var int */
+    private $maxTextLength = 8000;
+
+    /**
+     * VisibleTextLengthValidator constructor.
+     *
+     * @param int $maxTextLength Maximum length of text.
+     * @param FormatService $formatService Service to apply a formatter.
+     * @param LocaleInterface $locale For translating error messages.
+     */
+    public function __construct(int $maxTextLength, \Vanilla\Formatting\FormatService $formatService, \Vanilla\Contracts\LocaleInterface $locale) {
+        $this->locale = $locale;
+        $this->formatService = $formatService;
+        $this->maxTextLength = $maxTextLength;
+    }
+
+    /**
+     * Validate content length by stripping most meta-data and formatting.
+     *
+     * @param string $value User input content.
+     * @param string $field Field name where content is found.
+     * @param array $post POST array.
+     * @return mixed Either an Invalid Object or the value.
+     */
+    private function validate($value, $field, $post) {
+        $format = $post['Format'] ?? '';
+        $stringLength = $this->formatService->getVisibleTextLength($value, $format);
+        $diff = $stringLength - ($field->maxTextLength ?? $this->maxTextLength);
+        if ($diff <= 0) {
+            return $value;
+        } else {
+            $validationMessage = $this->locale->translate('ValidateLength' ?? '');
+            $fieldName = $this->locale->translate($field->Name ?? '');
+            return new Invalid(sprintf($validationMessage, $fieldName, abs($diff)));
+        }
+    }
+
+    /**
+     * Execute validate function for visible text validator.
+     *
+     * @param string $value User input content.
+     * @param string $field Field name where content is found.
```

Comment:
Same kinda thing as above. Is this supposed to be the name of the field or is it an object describing it?

Reply:
It's just the name.

Reply:
> It's just the name.

Are you sure about that? It looks like `$fieldInfo`, [an object containing the field's config info](https://github.com/vanilla/vanilla/blob/3eb6b83cc7c29d882318444eee49d689f033f508/library/core/class.validation.php#L796), is passed as [the second parameter](https://github.com/vanilla/vanilla/blob/3eb6b83cc7c29d882318444eee49d689f033f508/library/core/class.validation.php#L838) when this class is "invoked". It would _contain_ the name of the field (e.g. `$field->Name`).